test/integration/proxy_proto_integration_test.cc

```diff
@@ -26,7 +26,21 @@ TEST_P(ProxyProtoIntegrationTest, RouterRequestAndResponseWithBodyNoBuffer) {
   testRouterRequestAndResponseWithBody(1024, 512, false, &creator);
 }
 
-TEST_P(ProxyProtoIntegrationTest, RouterRequestAndResponseWithBodyNoBufferV6) {
+TEST_P(ProxyProtoIntegrationTest, v2RouterRequestAndResponseWithBodyNoBuffer) {
+  ConnectionCreationFunction creator = [&]() -> Network::ClientConnectionPtr {
+    Network::ClientConnectionPtr conn = makeClientConnection(lookupPort("http"));
+    constexpr uint8_t buffer[] = {0x0d, 0x0a, 0x0d, 0x0a, 0x00, 0x0d, 0x0a, 0x51, 0x55, 0x49,
```

Comment:
these are all well documented in the proxy proto spec, i'm not sure it adds value for me to parse them all out in comments here.

Reply:
You mean these examples are from the spec? Can you link to where to find them for reference?

Reply:
i mean all the fields (type, len, etc) are in the spec, pulling it all out in comments, will it really make it simpler?

Reply:
Ultimately, asking the next person that comes along that needs to understand this code/tests to go get the spec and convert the hex is a pretty poor experience and not great for group maintainability. I'm not going to block merge if you don't want to do it, but if it were me, I would do one of the following:
1) At minimum have some comments explaining a summary form of what options the byte buffer actually specifies.
2) Better, used some kind of packed or intermediate structure to describe what you want to test, then have a utility function to convert to bytes.

Reply:
i added some comments to explain what is the 'delta' from the normal header.
in practise there is only 3 here (v4/tcp, v6/tcp, non-inet) in here, w/ errors introduced. It would be a pain to write a parser to 'specify' what I want that also allows errors like illegal length etc.